### PR TITLE
scripts: Update generate_spirv.py for GPU-AV

### PR DESCRIPTION
**Pull request by sjfricke**
This is a clean-up to have a more consistent/cleaner way to generate shaders for GPU-AV as we might need more in the future

- Created a `layers/gpu_shaders` folder to hold the shaders 
- Moved the RTX AS computer shader into a file instead of embedded in the source
- Made shader constant file a single shared header
- updated `generate_spirv.py` and added instructions in the `README.md`
- cleaned up `gpu_pre_draw_vert` to be consistent with the coding style

### CODE REVIEW

File: layers/gpu_shaders/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# GPU Assisted Validation Shaders
+
+This directory is for holding shaders that are used for [GPU Assisted Validation](../../docs/gpu_validation.md). These are turned in SPIR-V when generating code with `generate_spirv.py`　and placed in a header file in [layers/generated](../generated/).
```
Comment (ncesario-lunarg) [inline layers/gpu_shaders/README.md:3]: `in` -> `into`?

File: layers/gpu_shaders/gpu_shaders_constants.h [hunk 2]
```diff
@@ -16,16 +16,20 @@
 //
 // * Author: Tony Barbour <tony@lunarg.com>
 //
-// Values used in the pre draw validation shader and then mapped and checked in gpu_validation
+// Values used netween the GLSL shaders and the GPU-VA logic
```
Comment (ncesario-lunarg) [inline layers/gpu_shaders/gpu_shaders_constants.h:19]: `GPU-VA` -> `GPU-AV`?

File: scripts/generate_spirv.py [hunk 3]
```diff
@@ -71,54 +63,93 @@ def compile(filename, tmpfile):
...
-for i in range(0, len(words), COLUMNS):
-    columns = ["0x%08x" % word for word in words[i:(i + COLUMNS)]]
-    literals.append(" " * INDENT + ", ".join(columns) + ",")
+    literals = []
+    for i in range(0, len(words), COLUMNS):
+        columns = ["0x%08x" % word for word in words[i:(i + COLUMNS)]]
+        literals.append(" " * INDENT + ", ".join(columns) + ",")
 
-header = """#include <stdint.h>
+    header = """#include <stdint.h>
```
Comment (ncesario-lunarg) [inline scripts/generate_spirv.py:81]: I'm personally a fan of the `f"""` strings as you can then read _inline_ what each `%s` represents, but not that's just me personally :)

**Review by ncesario-lunarg — APPROVED** [review]
LGTM if @Tony-LunarG thinks so!

File: layers/gpu_shaders/gpu_shaders_constants.h (hunk 2 above)
Comment (sjfricke) [reply layers/gpu_shaders/gpu_shaders_constants.h:19]: wow I have mixed that up a lot, will update!

File: layers/gpu_shaders/gpu_shaders_constants.h [hunk 4]
```diff
@@ -16,16 +16,20 @@
 //
 // * Author: Tony Barbour <tony@lunarg.com>
 //
-// Values used in the pre draw validation shader and then mapped and checked in gpu_validation
+// Values used netween the GLSL shaders and the GPU-AV logic
```
Comment (Tony-LunarG) [inline layers/gpu_shaders/gpu_shaders_constants.h:19]: netween -> between

File: scripts/generate_spirv.py [hunk 5]
```diff
@@ -71,54 +63,93 @@ def compile(filename, tmpfile):
 
         assert(words[0] == SPIRV_MAGIC)
 
-
     # remove temp file
     os.remove(tmpfile)
 
-    return (words, output.rstrip())
+    # The disassembled SPIR-V first line has file path to the shadre
```
Comment (Tony-LunarG) [inline scripts/generate_spirv.py:69]: shadre -> shader

File: layers/gpu_shaders/README.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# GPU Assisted Validation Shaders
+
+This directory is for holding shaders that are used for [GPU Assisted Validation](../../docs/gpu_validation.md). These are turned in SPIR-V when generating code with `generate_spirv.py`　and placed into a header file in [layers/generated](../generated/).
```
Comment (Tony-LunarG) [inline layers/gpu_shaders/README.md:3]: turned into?

**Review by Tony-LunarG — COMMENTED** [review]
Nice cleanup - just a couple comment typos

**Pull request merged**